api/v1beta1/solrcloud_types.go

```diff
@@ -576,8 +588,21 @@ const (
 )
 
 func (opts *ExternalAddressability) withDefaults(usesTLS bool) (changed bool) {
+	// TODO: Remove in v0.7.0
+	// If the deprecated IngressTLSTerminationSecret exists, use it to default the new location of the value.
+	// If that location already exists, then merely remove the deprecated option.
+	if opts.IngressTLSTerminationSecret != "" {
+		if !opts.HasIngressTLSTermination() {
+			opts.IngressTLSTermination = &SolrIngressTLSTermination{
+				TLSSecret: opts.IngressTLSTerminationSecret,
+			}
+		}
```

Comment:
[Q] If we receive a solrcloud that has both the deprecated IngressTLSTerminationSecret and an IngressTLSTermination object, should we log a warning or note the conflict in some other way?

I guess we wouldn't expect users to necessarily be checking the operator logs for that sort of thing.  So maybe there's no good way to surface this.  Just thinking aloud I guess...

Reply:
That's a good idea.

Reply:
Going to tackle this tomorrow. Everything else should be done though

Reply:
Done!